odoo/addons/base/controllers/avatar.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from hashlib import sha1
+
+from odoo.http import Controller, request, route, Response, local_redirect
+from odoo import tools
+
+# show avatar if no image or showAvatar 1
+# show image if image
+
+class Avatar(Controller):
+    @route(['/web/avatar/<int:user_id>'], auth='user', methods=['GET'])
+    def get_avatar(self, user_id=1):
+        if 'isPartner' in request.params and request.params['isPartner'] == '1':
+            user = request.env['res.users'].search([('partner_id', '=', user_id), ('groups_id.id', '=', request.env.ref('base.group_user').id)]).exists()
+        else:
+            user = request.env['res.users'].browse(user_id).exists()
```

Comment:
I'm not a fan of this route signature at all.

In particular this param named `user_id` that can become a `partner_id` if another param is set.
Also I don't understand the default value of 1. This field should be required and be given an explicit value if we want the avatar of someone.

So I would much prefer if the target model was explicit similarly to what we do in `/web/image` (with the `<string:model>/<int:id>` variant).

Also please explicit in the signature and document every other possible param such as `consume` which I'm not sure what it is supposed to do as it is.

Finally, it would be best to always keep `**kwargs` so that we are able to pass other arbitrary params, such as `debug=1`, `context` and other such values that are handled at another level in the system.